tests/common/fixtures/consistency_checker/query-asic.py

```diff
@@ -0,0 +1,426 @@
+        if status_code_range[0] <= status_code and status_code <= status_code_range[1]:
+            return status_str
+
+    return f"UNKNOWN_SAI_STATUS"
+
+
+def _get_attribute_value_from_asic(oid, attribute_oid):
+
+    oidType = pysairedis.sai_object_type_query(oid)
+    objectTypeName = pysairedis.sai_metadata_get_object_type_name(oidType).replace("SAI_OBJECT_TYPE_", "")
+    className = objectTypeName.lower()
+    if objectTypeName in ["BUFFER_POOL", "BUFFER_PROFILE"]:
+        className = "buffer"
+    api = getattr(pysairedis, f"sai_{className}_api_t")()
+    status = getattr(pysairedis, f"sai_get_{className}_api")(api)
+    assert status == 0, f"Failed to get sai API {api}. Status: {map_sai_status_to_str(status)} ({status})"
+
+    attr_metadata = pysairedis.sai_metadata_get_attr_metadata(oidType, attribute_oid)
+
+    attr = pysairedis.sai_attribute_t()
+    attr.id = attribute_oid
+    if attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT32_LIST:
+        attr.value.u32list.count = 32
+        attr.value.u32list.list = pysairedis.new_uint32_t_arr(attr.value.u32list.count)
+
+    status = getattr(api, f"get_{objectTypeName.lower()}_attribute")(oid, 1, attr)
+    assert status == 0, (f"Failed to call SAI API get_{objectTypeName.lower()}_attribute for oid {oid} and attribute "
+                         f"{attribute_oid}. Status: {map_sai_status_to_str(status)} ({status})")
+
+    if attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_BOOL:
+        attr_value = attr.value.booldata
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT8:
+        attr_value = attr.value.u8
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_INT8:
+        attr_value = attr.value.s8
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT16:
+        attr_value = attr.value.u16
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_INT16:
+        attr_value = attr.value.s16
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT32:
+        attr_value = attr.value.u32
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_INT32:
+        attr_value = attr.value.s32
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT64:
+        attr_value = attr.value.u64
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_INT64:
+        attr_value = attr.value.s64
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_OBJECT_ID:
+        attr_value = attr.value.oid
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_UINT32_LIST:
+        attr_value = attr.value.u32list
+    elif attr_metadata.attrvaluetype == pysairedis.SAI_ATTR_VALUE_TYPE_MAC:
+        # The following mac address is a pointer to a 6 element array of uint8_t
+        # there doesn't seem to be a way to get the value from the pointer using pysairedis
```

Comment:
Good catch. To fix this, `src/sonic-sairedis/pyext/pysairedis.i` would need to be told to generate pointer functions for `uint8_t`. There's existing code for `uint32_t`, that just needs to be copied for `uint8_t`.

Reply:
I've just pushed https://github.com/sonic-net/sonic-mgmt/pull/14402/commits/378d163469a38f1b33d050af2cff5ffda1c553c1 with a fix that should mean we don't need to add the additional uint8_t bindings. Let me know what you think.